src/net/download.rs

```diff
@@ -1696,65 +1746,58 @@ impl PeerNetwork {
     /// create the HTTP request.  Pops requests off the front of request_keys, and returns once it successfully
     /// sends out a request via the HTTP peer.  Returns the event ID in the http peer that's
     /// handling the request.
-    fn begin_request<F>(
+    pub fn begin_request<T: Requestable + std::fmt::Display>(
         network: &mut PeerNetwork,
         dns_lookups: &HashMap<UrlString, Option<Vec<SocketAddr>>>,
-        request_name: &str,
-        request_keys: &mut VecDeque<BlockRequestKey>,
+        requestables: &mut VecDeque<T>,
```

Comment:
Please restore `request_name` -- I use it in debugging the block downloader.

Reply:
I did pay a close attention to the logs that this function was previously printing, and there's no loss of informations - the request_name (derived from kind) is already is included in the `impl Display` of a BlockRequestKey.